### PR TITLE
feat(cdk): async framework with dependency injection

### PR DESCRIPTION
**Pull request by jdpgrailsdev**

### CODE REVIEW

**Comment by jdpgrailsdev** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/airbytehq/airbyte/35990?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#35990** <a href="https://app.graphite.dev/github/pr/airbytehq/airbyte/35990?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#35933** <a href="https://app.graphite.dev/github/pr/airbytehq/airbyte/35933?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#35835** <a href="https://app.graphite.dev/github/pr/airbytehq/airbyte/35835?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#35833** <a href="https://app.graphite.dev/github/pr/airbytehq/airbyte/35833?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#35832** <a href="https://app.graphite.dev/github/pr/airbytehq/airbyte/35832?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#35829** <a href="https://app.graphite.dev/github/pr/airbytehq/airbyte/35829?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#35826** <a href="https://app.graphite.dev/github/pr/airbytehq/airbyte/35826?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#35823** <a href="https://app.graphite.dev/github/pr/airbytehq/airbyte/35823?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#35821** <a href="https://app.graphite.dev/github/pr/airbytehq/airbyte/35821?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @jdpgrailsdev and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

**Comment by stephane-airbyte** [comment]
I feel like this is a very important PR. We're actually introducing DI, which is something central to what we're trying to achieve in this stack. Unfortunately, this PR seems to be mixing DI introduction with a bunch of code cleanup, making essentially unreviewable

File: airbyte-cdk/java/airbyte-cdk/core/src/test/kotlin/io/airbyte/cdk/integrations/destination/async/DetectStreamToFlushTest.kt [hunk 1]
```diff
@@ -8,143 +8,129 @@ import edu.umd.cs.findbugs.annotations.SuppressFBWarnings
...
     private lateinit var flusher: DestinationFlushFunction
+    private lateinit var runningFlushWorkers: RunningFlushWorkers
 
     @BeforeEach
     internal fun setup() {
-        flusher = Mockito.mock(DestinationFlushFunction::class.java)
-        Mockito.`when`(flusher.optimalBatchSizeBytes).thenReturn(SIZE_200MB)
+        bufferDequeue = mockk()
+        flusher = mockk()
+        runningFlushWorkers = mockk()
```
Comment (gisripa) [inline airbyte-cdk/java/airbyte-cdk/core/src/test/kotlin/io/airbyte/cdk/integrations/destination/async/DetectStreamToFlushTest.kt:39]: Did we decide on continuing with mockito in initial conversion phase or 
Is the intention to slowly start introducing mockk for DI'd classes for new path ?

**Review by gisripa — APPROVED** [review]
Sooner or later, we need to decouple this `TypingDedupingFlag.isDestinationV2` implicit dependency generated somewhere up the callstack from JsonNode to proper DI.

File: airbyte-cdk/java/airbyte-cdk/core/src/main/kotlin/io/airbyte/cdk/core/command/option/AirbyteConfiguredCatalog.kt [hunk 2]
```diff
@@ -18,10 +18,10 @@ import io.micronaut.context.annotation.ConfigurationProperties
     value = ["NP_NONNULL_RETURN_VIOLATION"],
     justification = "Uses dependency injection",
 )
-class AirbyteConfiguredCatalog {
+open class AirbyteConfiguredCatalog {
     lateinit var json: String
 
-    fun getConfiguredCatalog(): ConfiguredAirbyteCatalog {
+    open fun getConfiguredCatalog(): ConfiguredAirbyteCatalog {
```
[the remark this answers is not in the document]
Comment (jdpgrailsdev) [reply airbyte-cdk/java/airbyte-cdk/core/src/main/kotlin/io/airbyte/cdk/core/command/option/AirbyteConfiguredCatalog.kt:24]: @postamar Temporarily yes.  This is to allow existing Java code that uses this class, but does use DI yet, can create a different anonymous implementation that just passes through to the raw JSON node that is still be used in the unconverted code.

**Review by jbfbell — APPROVED** [review]

**Pull request closed without merging**